File: Dataset/Leetcode/train/98/114.cpp
 class Solution {
public:
    bool XXX(TreeNode* root) {
        if(!root || (!root->left && !root->right))
            return true;
        if(!root->left)
            return (getMin(root->right) > root->val) && XXX(root->right);
        else if(!root->right)
            return (getMax(root->left) < root->val) && XXX(root->left);
        else 
            return (getMin(root->right) > root->val && getMax(root->left) < root->val) && XXX(root->left) && XXX(root->right);
    }

    int getMin(TreeNode *node) {
        if(!node->left)
            return node->val;
        return getMin(node->left);
    }

    int getMax(TreeNode *node) {
        if(!node->right)
            return node->val;
        return getMax(node->right);
    }
};

File: Dataset/Leetcode/valid/69/114.java
class Solution {
    public int XXX(int x) {
        if(x==0){
            return 0;
        }
        long left=1;
        long right=x;
        while(right>=left){
            long mid=(left+right)/2;
            if(mid*mid<=x&&(mid+1)*(mid+1)>x){
                return (int)mid;
            }else  if(mid*mid<x){
                left=mid+1;
            }else{
                right=mid-1;
            }
        }
        return -1;
    }
}

